test-suite/src/test/java/io/micronaut/kafka/docs/seek/rebalance/ProductListener.java

```diff
@@ -0,0 +1,47 @@
+package io.micronaut.kafka.docs.seek.rebalance;
+
+import io.micronaut.configuration.kafka.ConsumerAware;
+import io.micronaut.configuration.kafka.annotation.*;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.kafka.docs.Product;
+import io.micronaut.core.annotation.NonNull;
+import jakarta.inject.Inject;
+import org.apache.kafka.clients.consumer.*;
+import org.apache.kafka.common.TopicPartition;
+import java.util.*;
+
+@KafkaListener(offsetReset = OffsetReset.EARLIEST)
+@Requires(property = "spec.name", value = "ConsumerRebalanceListenerTest")
+public class ProductListener implements ConsumerRebalanceListener, ConsumerAware {
+
+    List<Product> processed = new ArrayList<>();
+    private Consumer consumer;
+
+    @Inject
+    public ProductListener(ProductListenerConfiguration config) {
```

Comment:
why the empty constructor?

Reply:
The empty constructor is there because I need `ProductListenerConfiguration` to produce the test products _before_ `ProductListener` rebalances. This dependency ensures that things happen in a deterministic order.